L1Trigger/L1TMuon/src/MuonTriggerPrimitive.cc

```diff
@@ -478,7 +495,11 @@ void TriggerPrimitive::print(std::ostream& out) const {
       out << "Pad high      : " << _gem.pad_hi << std::endl;
       break;
     case kME0:
-      out << detId<ME0DetId>() << std::endl;
+      try {
+        out << detId<ME0DetId>() << std::endl;
+      } catch (...) {
+        out << detId<GEMDetId>() << std::endl;
+      }
```

Comment:
@watson-ij @omiguelc It's been a while, and there is still this catch here that is not allowed under CMSSW rules. I would like to unstick this PR and finalize it, since everything else seems to be in order. Since this is just a print statement, and not strictly needed for the running of the module, I would suggest just pulling this statement until something better can be done:

```suggestion
```

Reply:
Hi @aloeliger @omiguelc sorry I missed this one. The exception will be thrown from the ME0DetId constructor as a cms::Exception("InvalidDetId"). Is it sufficient to change it to
```c++
        } catch (cms::Exception& e) {
```
?

Reply:
@watson-ij CMSSW very strictly discourages using exceptions for control flow (which I am frequently reminded when I try do it). This I think unfortunately falls under that category. Either there should be a proper case for GEM vs ME0 ID printing, or this should be removed from CMSSW until there is.

Reply:
Ah. Okay, in that case @omiguelc this should be a solution:
```c++
      if (detId<DetId>().subdetId() == MuonSubdetId::ME0) {
        out << detId<ME0DetId>() << std::endl;
      } else {
         out << detId<GEMDetId>() << std::endl;
      }
```